### PR TITLE
ext: lib: crypto: Generic mbedtls config file

### PR DESCRIPTION
**Pull request by rlubos**
Part of #7118.

This PR adds a generic config file for mbedTLS, that can be altered with Kconfig.

The idea behind this is to simplify TLS configuration for end-user. User can use Kconfig to select supported TLS algorithms, and the config file will automatically satisfy dependencies. It has been used with #7118, yet for this PR it has been freed from secure socket dependencies and improved a little bit.

For configs not covered by the generic config file (e. g. buffer sizes), a user can specify user config, that will be included at the end of generic config (similar behaviour as in the template config provided with mbedTLS).

The default configuration for `config-tls-generic.h` mimics the configuration of `config-mini-tls1_2.h`, hence it was set as a default config file. Tested with `echo_client`/`echo_server` samples with corresponding `net-tools` utils.

### CODE REVIEW

**Review by jukkar — APPROVED** [review]
LGTM

**Review by pfalcon — APPROVED** [review]
I very much appreciate doing this.

I assume this is formally independent from https://github.com/zephyrproject-rtos/zephyr/pull/8814 and can be merged ahead of it (as a hint for project maintainers).

File: ext/lib/crypto/mbedtls/Kconfig.tls-generic [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+endmenu
+
+config TLS_PEM_CERTIFICATE_FORMAT
+	bool "Enable support for PEM certificate format"
+	default n
+	help
+	  By default only DER (binary) format of certificates is supported. Enable
+	  this option to enable support for PEM format.
+
+config TLS_USER_CONFIG_ENABLE
```
Comment (pfalcon) [inline ext/lib/crypto/mbedtls/Kconfig.tls-generic]: Did you consider having just a single Kconfig options for this? E.g. if  TLS_USER_CONFIG_FILE is defined to non-empty value, #include it?

Comment (rlubos) [reply ext/lib/crypto/mbedtls/Kconfig.tls-generic]: Yes, that was my initial thought, but that would require to check for an empty string in preprocessor, and I didn't find an easy way to do that. Any suggestions on this?

**Comment by rlubos** [comment]
@pfalcon Yes, there are no dependencies between these PRs.

File: ext/lib/crypto/mbedtls/Kconfig.tls-generic (hunk 1 above)
Comment (pfalcon) [reply ext/lib/crypto/mbedtls/Kconfig.tls-generic]: Yeah, I never thought what `#if ""` would evaluate too, but it's actually an error to use strings in #if at all:

~~~
a.c:1:11: error: token ""foo"" is not valid in preprocessor expressions
 #define a "foo"
           ^
a.c:3:5: note: in expansion of macro ‘a’
 #if a
     ^
~~~

So, nevermind.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**